packages/react/src/components/ComboBox/ComboBox.tsx

```diff
@@ -89,23 +266,21 @@ const ComboBox = React.forwardRef((props, ref) => {
     light,
     onChange,
     onInputChange,
-    onToggleClick, // eslint-disable-line no-unused-vars
+    onToggleClick, 
     placeholder,
     readOnly,
     selectedItem,
     shouldFilterItem,
     size,
     titleText,
     translateWithId,
-    type, // eslint-disable-line no-unused-vars
     warn,
     warnText,
-    onStateChange, // eslint-disable-line no-unused-vars
```

Comment:
Typically these are destructured out of `props` so that when `...rest` is spread onto a child element it won't cause errors trying to apply attributes to an element that doesn't support them. 

Can you confirm adding `type` and/or `onStateChange` onto the component does not produce errors in the browser console?

Reply:
I didn't realize that could be an issue and checking that further shows that adding `onStateChange` would cause an error in that case. I've added these props back in to avoid this.